Taking sleep timeout as parameter

tahir24434 opened the pull request:
Signed-off-by: trauf tahir@plumgrid.com

Commit:
Function for checking that tasks have been distributed evenly

Signed-off-by: trauf <tahir@plumgrid.com>

tahir24434 committed:
Merge branch 'master' into master

sushilks commented on src/main/python/hydra/lib/hydrabase.py:
```diff
@@ -473,6 +473,35 @@ def refresh_app_info(self, name):
+
+        l.info(task_map)
+        # First check that atleast every host has been engaged.
+        if len(task_map) != num_hosts:
```
I would like the function to handle or work for all cases, 
It doest not work for the case if "num_hosts" = 3 and the number of instances of app is 1 or 5. (Not a multiple of 3).

Please make sure it's handling non-multiples as well allows the im-balance to as close to perfect.
i.e. if num_hosts = 3 and instances = 4 a valid distribution can be 
Host1 = 1 instance
Host2 = 1 instance
Host3 = 2 instance.

You can also choose to return the type of imblance that you see "Perfect Balance", "UNEven Balance", "Not Balanced".

Commit:
Taking sleep timeout as parameter

In case of large number of tasks, the retries are not sufficient with 1 second sleep.

Signed-off-by: trauf <tahir@plumgrid.com>
